tests/provision/become/container/test.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+. /usr/share/beakerlib/beakerlib.sh || exit 1
+
+rlJournalStart
+    rlPhaseStartSetup
+        rlRun "pushd data"
+        # Try several times to build the container
+        # https://github.com/teemtee/tmt/issues/2063
+        build="podman build -t become-container-test:latest ."
+        rlRun "rlWaitForCmd '$build' -m 5 -d 5" || rlDie "Unable to prepare the image"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Container, test with become=true"
+        rlRun "tmt run -rvvv plan --name /test/root"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Container, test with become=false"
+        rlRun "tmt run -rvvv plan --name /test/user"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Container, prepare/finish inline with become=true"
+        rlRun "tmt run -rvvv plan --name /prepare-finish/root/inline"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Container, prepare/finish inline with become=false"
+        rlRun "tmt run -rvvv plan --name /prepare-finish/user/inline"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Container, prepare/finish scripts with become=true"
+        rlRun "tmt run -rvvv plan --name /prepare-finish/root/scripts"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Container, prepare/finish scripts with become=false"
+        rlRun "tmt run -rvvv plan --name /prepare-finish/user/scripts"
+    rlPhaseEnd
+
+    rlPhaseStartCleanup
+        rlRun "popd"
+        rlRun "podman image rm -f localhost/become-container-test:latest" 0 "Remove custom image"
+    rlPhaseEnd
+rlJournalEnd
```

Comment:
Thank you @psss  for the feedback. When  I did it it was more than just branching within the tests, but coexisting both ways of provisioning within the same `data/` folder at the root level. But I'll look at it again and see if I can come up with something.

Reply:
Thanks! You can also have a look at #2688 where I'm working on simplifying the provision methods handling. The `/tests/core/escaping` test [demonstrates](https://github.com/teemtee/tmt/pull/2688/files#diff-2c66578111b7bcb4fb55c0969108e90fa1626c6534d1fd12acfa1acd96c56451) how it will work in the simplified form.

Reply:
@psss , I just pushed a new commit deduplicating code. I got it to work. Let me know how it looks like.

After the review and the checks passes, I was thinking of squashing all into one commit to keep the history clean with just one commit if that's OK.

Reply:
Thanks for the improvements! I've squashed your changes and rebased on the latest `main`. Should be ready for merge if tests are green.